flow360/component/simulation/translator/volume_meshing_translator.py

```diff
@@ -160,22 +193,30 @@ def get_volume_meshing_json(input_params: SimulationParams, mesh_units):
 
     ##:: Step 3: Get volumetric global settings
     translated["volume"] = {}
-    # firstLayerThickness can be locally overridden so after completeness check, we can
-    # get away with the first instance's value.
-    translated["volume"]["firstLayerThickness"] = get_global_setting_from_per_item_setting(
-        meshing_params.refinements,
-        BoundaryLayer,
-        "first_layer_thickness",
-        allow_get_from_first_instance_as_fallback=True,
-    ).value.item()
-    # growthRate can only be global so after completeness check, and use same logic as curvature_resolution_angle.
-    translated["volume"]["growthRate"] = get_global_setting_from_per_item_setting(
-        meshing_params.refinements,
-        BoundaryLayer,
-        "growth_rate",
-        allow_get_from_first_instance_as_fallback=True,
-        # `allow_get_from_first_instance_as_fallback` can be true because completeness check is done in Params
-    )
+
+    if meshing_params.defaults.boundary_layer_first_layer_thickness is None:
+        # `first_layer_thickness` can be locally overridden so after completeness check, we can
+        # get away with the first instance's value if global one does not exist.
+        default_first_layer_thickness = get_attribute_from_instance_list(
+            meshing_params.refinements,
+            BoundaryLayer,
+            "first_layer_thickness",
+            only_find_when_entities_none=False,
+        )
+    else:
+        default_first_layer_thickness = meshing_params.defaults.boundary_layer_first_layer_thickness
+
+    translated["volume"]["firstLayerThickness"] = default_first_layer_thickness.value.item()
+
+    # growthRate can only be global
+    if meshing_params.defaults.boundary_layer_growth_rate is None:
+        raise Flow360TranslationError(
+            "`boundary_layer_growth_rate` is not set.",
+            None,
+            ["meshing", "volume_zones"],
```

Comment:
I think this path is not correct, it should be ["meshing", "defaults"]?

Reply:
Right, corrected.